src/main/java/com/nicoislost/ZoomO.java

```diff
@@ -28,197 +45,154 @@ public void onInitializeClient() {
-						zoomOrder[i] = zoomOrder[i + 1];
-					}
-				}
-				return zoomOrder[0];
-			}
-		} else { // All three keys are pressed.
-			if (zoomOrder[2] == 0) {
-				// First time pressing the third key.
-				try {
-					zoomOrder[2] = 6 / (zoomOrder[0] * zoomOrder[1]); //TODO find a better way to avoid dividing by zero.
-				} catch (Exception e) {
-					zoomOrder[2] = 0;
-				}
-			}
-			return zoomOrder[0];
 		}
-		return 0;
 	}
 
 	/**
-	 * Translates the number output from {@link #zoomModifierNum()} to a double with
-	 * the zoom amount defined in the config.
-	 * 
-	 * @param keyNum - 1, 2, or 3
-	 * @return A double that can be interpreted as a FOV value in the source based
-	 *         on corresponding values in the config. Returns 0 if keyNum is
-	 *         invalid.
+	 * @return Returns the currently active key binding or null if none are active
 	 */
-	public static double ZoomModifier(int keyNum) {
-		if (keyNum == 1) {
-			return CONFIG.Zoom1();
-		} else if (keyNum == 2) {
-			return CONFIG.Zoom2();
-		} else if (keyNum == 3) {
-			return CONFIG.Zoom3();
+	private static KeyBind getCurrentKeyBind() {
+		if (!isZooming() || ACTIVE_KEYS.isEmpty()) {
+			return null;
 		}
-		return 0;
+		return ACTIVE_KEYS.get(0);
 	}
 
 	/**
-	 * Tells us how many keybindings are pressed to use in
-	 * {@link #zoomModifierNum()}
-	 * 
-	 * @return 0-3 depending on how many keys are pressed.
+	 * Returns the configured zoom modifier for the currently active key binding
+	 * @return The configure zoom modifier value for the active key binding or 0 if no key binding is active
 	 */
-	public static int keysPressed() {
-		//This converts our booleans into ints and sums them up.
-		return (key1() ? 1 : 0) + (key2() ? 1 : 0) + (key3() ? 1 : 0);
+	private static double getCurrentZoomModifier() {
+		KeyBind keyBind = getCurrentKeyBind();
+		if (keyBind == null) {
+			return 0;
+		}
+		LOGGER.info("Current key = " + keyBind);
+		return keyBind.getConfigValue();
 	}
 
 	/**
-	 * @return 1 if 0 or 1 keys are pressed, 2 if 2 keys are pressed, 6 if 3 keys
-	 *         are pressed.
+	 * Delegates the functionality of the MixIn callback
+	 * @param callbackInfo
 	 */
-	public static int keyNum() {
-		int num = 1;
-		// We start with one to avoid multiplying by zero, and it is only activates if
-		// at least 1 key is pressed.
+	// One thing to remember is, this is going to be called ALOT while one of the key bindings is being pressed,
+	// so we should be conscious of doing as little work as fast as possible - so, caching values if possible and/or
+	// reducing the amount of computational work we're doing.
+	// One thing we could do is check the previous state with the current state and if nothing's changed, returned
+	// the previously calculated value, but, arguably, comparing two `List` isn't the fastest thing in the world either,
+	// so maybe a `BitSet` instead?
+	public static void getZoomLevel(CallbackInfoReturnable<Double> callbackInfo) {
+		syncKeys();
```

Comment:
This could/should be renamed, as it's not actually returning anything, but I maintained the naming between this the `ZoomOMixin`